sympy/functions/elementary/hyperbolic.py

```diff
@@ -230,13 +230,17 @@ def _eval_rewrite_as_coth(self, arg, **kwargs):
         return 2*coth_half/(coth_half**2 - 1)
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        from sympy import Order
-        arg = self.args[0].as_leading_term(x)
+        arg = self.args[0].as_leading_term(x, logx=logx, cdir=cdir)
+        arg0 = arg.subs(x, 0)
 
-        if x in arg.free_symbols and Order(1, x).contains(arg):
+        if arg0 is S.NaN:
+            arg0 = arg.limit(x, 0, dir='-' if cdir == -1 else '+')
```

Comment:
`cdir == -1` is too restrictive. `cdir`, when recomputed by `dir`, may have any value. `t*cdir` denotes the same direction for all positive values of `t`. In principle, `cdir` could be normalized to have absolute value `1` but then the expression would often have `sqrt` expressions in the denominator. In this case, a proper condition would be `cdir.is_negative`.